osf/models/preprint_service.py

```diff
@@ -74,11 +74,25 @@ def deep_url(self):
 
     @property
     def url(self):
-        return '/{}/'.format(self._id)
+        if self.provider._id == 'osf' or self.provider.domain:
+            return '/{}/'.format(self._id)
+
+        return '/preprints/{}/{}/'.format(self.provider._id, self._id)
+
+    if settings.DEV_MODE:
```

Comment:
If we add another setting that sounds semantically the same as DEV_MODE, that will cause confusion. My preference would be not to require a new mode, but if we do, then it should be something like "ENABLE_PROVIDER_DOMAINS".